roles/edpm_pre_adoption_validation/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Validate that hostname for the EDPM openstack services do not change
+  become: true
+  tags:
+    - adoption
+    - validation
+  block:
+    - name: Check if old nova.conf exists
+      ansible.builtin.stat:
+        path: "{{ edpm_pre_adoption_validation_old_nova_config }}"
+      register: nova_old_config_file
+
+    - name: Compare old nova service host to {{ canonical_hostname }}
+      when: nova_old_config_file.stat.exists
+      ansible.builtin.shell: |
+        set -o pipefail
+        egrep -e '^host.*=' {{ edpm_pre_adoption_validation_old_nova_config }} | cut -d '=' -f2 | sed 's/ //g'
+      register: nova_old_config_output
+      changed_when: true
+      failed_when: nova_old_config_output.stdout != canonical_hostname
+
+    - name: Check if old neutron.conf exists
+      ansible.builtin.stat:
+        path: "{{ edpm_pre_adoption_validation_old_neutron_config }}"
+      register: neutron_old_config_file
+
+    - name: Compare old neutron service host to {{ canonical_hostname }}
+      when: neutron_old_config_file.stat.exists
+      ansible.builtin.shell: |
+        set -o pipefail
+        egrep -e '^host.*=' {{ edpm_pre_adoption_validation_old_neutron_config }} | cut -d '=' -f2 | sed 's/ //g'
+      register: neutron_old_config_output
+      changed_when: true
+      failed_when: neutron_old_config_output.stdout != canonical_hostname
+
+    - name: Print validated information
+      ansible.builtin.debug:
+        msg:
+          - "canonical_hostname: {{ canonical_hostname }}"
+          - "old nova service host config: {{ nova_old_config_output.stdout | default('not found')}}"
+          - "old neutron service host config: {{ neutron_old_config_output.stdout | default('not found')}}"
```

Comment:
ok i guess this works
im slightly inclined to say  we could take the config lookup, assert and printing and execat them into a separate task file

then we could include that in a loop but since we are only checking two files I'm fine with keeping this simple and duplicating the check

if we were to do this for  more files i would like to exctart this into reuabel code but for now I'm ok with the approach you have taken.

Reply:
yeah, I thought about a loop. It is sad that ansible cannot directly loop on a block.